### PR TITLE
Added CLI program

### PR DESCRIPTION
**Pull request by lukewestby**
Let me know what you think! `./elm-css.js --help` should print the usage. I also flattened out some of the Promise code in the API impl to reduce some of the Promise creation overhead a little. Btw, I love all the recent changes :smiley:

### CODE REVIEW

**Comment by rtfeldman** [comment]
Nice! I added a super basic binary in there earlier today, but this looks much more fully featured.

File: elm-css.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  .version(pkg.version)
+  .option('-r, --root [projectDir]', 'root directory of the project')
+  .option('-s, --source [sourcePath]', 'path to the source file for your stylesheets module')
+  .option('-o, --output [outputDir]', 'directory in which to write CSS file')
+  .option('-m, --module [moduleName]', '(optional) name of stylesheets module in your project', null, 'Stylesheets')
+  .option('-p, --port [portName]', '(optional) name of the port from which to read CSS results', null, 'files')
+  .parse(process.argv);
+
+var requiredArgNames = [
+  'root', 'source', 'output'
```
Comment (rtfeldman) [inline elm-css.js]: I think it's reasonable to have `root` default to `process.cwd()`

File: elm-css.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+elmCss(
+  program.root,
+  program.source,
+  program.output,
+  program.module,
+  program.port
+).then(function (results) {
+  console.log(chalk.green('Successfully generated output! The following css files were createTmpDir: '));
+  results.forEach(function (result) {
+    console.log(chalk.blue('- ' + result.filename));
```
Comment (rtfeldman) [inline elm-css.js]: :heart_eyes:

File: index.js [hunk 3]
```diff
@@ -6,79 +6,104 @@ var jsEmitterFilename = "emitter.js";
...
+  return function (result) {
+    return new Promise(function(resolve, reject) {
+      fs.writeFile(path.join(outputDir, result.filename), result.content + "\n", function(err, file) {
+        return err ? reject(err) : resolve(result);
+      });
+    });
+  }
+}
+
+
```
Comment (rtfeldman) [inline index.js]: :+1:

File: index.js [hunk 4]
```diff
@@ -6,79 +6,104 @@ var jsEmitterFilename = "emitter.js";
...
+      });
+
+      return failures.length > 0
+      ? reject(reportFailures(failures))
+      : resolve(stylesheets);
+    });
+  };
+}
+
+function makeRequirable(dest) {
```
Comment (rtfeldman) [inline index.js]: Sweet refactor! :smiley_cat:

**Comment by rtfeldman** [comment]
This is awesome. Thanks for the excellent work!

File: elm-css.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+})
+
+elmCss(
+  program.root,
+  program.source,
+  program.output,
+  program.module,
+  program.port
+).then(function (results) {
+  console.log(chalk.green('Successfully generated output! The following css files were createTmpDir: '));
```
Comment (lukewestby) [inline elm-css.js]: `createTmpDir` darn autocomplete :anger:

Comment (rtfeldman) [inline elm-css.js]: haha it's all good - fixed in `master`

**Pull request merged**